pyeapi/api/interfaces.py

```diff
@@ -394,8 +394,12 @@ def create(self, name):
             NotImplementedError: creating Ethernet interfaces is not supported
 
         """
+        ver = self.node.version_number.split('.')
+        if int(ver[0]) >= 4 and int(ver[1]) >= 15:
+            return self.configure('interface %s' % name)
```

Comment:
I personally wouldn't do version checking.  In this case it's pretty easy to draw a line in the sand and determine which releases support subinterface creation, but I would just pass the command to EOS and let it provide the error message.